src/Symfony/Component/HttpFoundation/HeaderUtils.php

```diff
@@ -193,6 +193,64 @@ public static function makeDisposition(string $disposition, string $filename, st
         return $disposition.'; '.self::toString($params, ';');
     }
 
+    /**
+     * Like parse_str(), but preserves dots in variable names.
+     */
+    public static function parseQuery(string $query, bool $ignoreBrackets = false, string $separator = '&'): array
```

Comment:
Shouldn't we just use the native `parse_str()` if `$ignoreBracket` is set to `false`? And so rename `
`$ignoreBrackets` to something else such as `$phpCompat = true` or something like that?

Many PHP libraries rely on the fact that dot are replaced by underscores and we may introduce issues by changing this. I would prefer to have a pure PHP mode (with all the oddities, including the dots etc), and a "strict" mode doing nothing more than the `URL` class in JS for instance. It feels wrong to me to have some intermediary modes such as "don't replace the dots but ignore brackets".

Reply:
I'm sorry I'm missing your logic. To me, it feels wrong to parse dots (replacing them with `_`) but ignore brackets.
The purpose of the method is to give you access to the original names and have a toggle to accept multiple values with or without using `[]`.
Libs that expect keys mangled by php should be given arrays mangled by php, ie you wouldn't use this function with them.